e2e/utils/waits-utils.ts

```diff
@@ -418,17 +400,52 @@ export async function waitWhileLoading(
       confidenceCounter = confidenceCounter > 0 ? confidenceCounter-- : 0;
       error = err;
     }
-    if (maxRetries <= 0) {
-      logger.error(`ERROR: Loading spinner has not stopped. Spinner css is "${spinElementsCss}"`);
-      logger.error(error.stack);
-      await takeScreenshot(page, makeDateTimeStr('ERROR_Spinner_Timeout'));
-      throw new Error(error.message);
+
+    const spentTime = Date.now() - startTime;
+    if (spentTime > maxTime) {
+      if (confidenceCounter === 0) {
+        logger.error(`ERROR: Loading spinner has not stopped. Spinner css is "${spinElementsCss}"`);
+        logger.error(error.stack);
+        await takeScreenshot(page, makeDateTimeStr('ERROR_Spinner_TimeOut'));
+        throw new Error(error.message);
+      }
+      logger.info('Waiting for loading spinner to stop has exceeded maximum wait time. Test will continue.');
+      await takeScreenshot(page, makeDateTimeStr('Spinner_TimeOut'));
+      return;
     }
-    maxRetries--;
-    await page.waitForTimeout(200).then(waitUntilSpinnerGone); // short pause then retry
+
+    await page.waitForTimeout(200); // short pause then retry
+    await waitUntilSpinnerGone(maxTime - spentTime); // unused time
   };
 
-  await waitUntilSpinnerGone();
+  await waitUntilSpinnerGone(timeout); // Wait do not exceed timeout
+}
+
+async function assertValidPage(page: Page, timeout: number): Promise<boolean> {
```

Comment:
Please document what the return value means here

Reply:
Done.